packages/components/src/unit-control/index.tsx

```diff
@@ -92,24 +106,24 @@ function UnitControl(
 		setUnit( next );
 	};
 
-	const mayUpdateUnit = ( event ) => {
-		if ( ! isNaN( event.target.value ) ) {
+	const mayUpdateUnit = ( event: SyntheticEvent< HTMLInputElement > ) => {
+		if ( ! isNaN( Number( event.currentTarget.value ) ) ) {
```

Comment:
TypeScript complains about passing a string to `isNaN` so here we're explicitly trying to coerce to a number before doing the `isNaN` check. I believe this doesn't change the behaviour in practice, but happy to investigate further if it doesn't look right.

Reply:
It does look good to me 👍